scripts/datasets/pretrain_corpus/prepare_wikipedia.py

```diff
@@ -0,0 +1,181 @@
+    "es", "et", "eu", "ext", "fa", "ff", "fi", "fiu-vro", "fj", "fo", "fr",
+    "frp", "frr", "fur", "fy", "ga", "gag", "gan", "gd", "gl", "glk", "gn",
+    "gom", "gor", "got", "gu", "gv", "ha", "hak", "haw", "he", "hi", "hif",
+    "ho", "hr", "hsb", "ht", "hu", "hy", "ia", "id", "ie", "ig", "ii",
+    "ik", "ilo", "inh", "io", "is", "it", "iu", "ja", "jam", "jbo", "jv", "ka",
+    "kaa", "kab", "kbd", "kbp", "kg", "ki", "kj", "kk", "kl", "km", "kn", "ko",
+    "koi", "krc", "ks", "ksh", "ku", "kv", "kw", "ky", "la", "lad", "lb",
+    "lbe", "lez", "lfn", "lg", "li", "lij", "lmo", "ln", "lo", "lrc", "lt",
+    "ltg", "lv", "mai", "map-bms", "mdf", "mg", "mh", "mhr", "mi", "min", "mk",
+    "ml", "mn", "mr", "mrj", "ms", "mt", "mus", "mwl", "my", "myv", "mzn", "na",
+    "nah", "nap", "nds", "nds-nl", "ne", "new", "ng", "nl", "nn", "no", "nov",
+    "nrm", "nso", "nv", "ny", "oc", "olo", "om", "or", "os", "pa", "pag", "pam",
+    "pap", "pcd", "pdc", "pfl", "pi", "pih", "pl", "pms", "pnb", "pnt", "ps",
+    "pt", "qu", "rm", "rmy", "rn", "ro", "roa-rup", "roa-tara", "ru", "rue",
+    "rw", "sa", "sah", "sat", "sc", "scn", "sco", "sd", "se", "sg", "sh", "si",
+    "simple", "sk", "sl", "sm", "sn", "so", "sq", "sr", "srn", "ss", "st",
+    "stq", "su", "sv", "sw", "szl", "ta", "tcy", "te", "tet", "tg", "th", "ti",
+    "tk", "tl", "tn", "to", "tpi", "tr", "ts", "tt", "tum", "tw", "ty", "tyv",
+    "udm", "ug", "uk", "ur", "uz", "ve", "vec", "vep", "vi", "vls", "vo", "wa",
+    "war", "wo", "wuu", "xal", "xh", "xmf", "yi", "yo", "za", "zea", "zh",
+    "zh-classical", "zh-min-nan", "zh-yue", "zu"]
+
+_BASE_URL_TMPL = "https://dumps.wikimedia.org/{lang}wiki/{date}/{lang}wiki-{date}-pages-articles.xml.bz2"
+_CURR_DIR = os.path.realpath(os.path.dirname(os.path.realpath(__file__)))
+
+
+def get_url(lang, date):
+    return _BASE_URL_TMPL.format(lang=lang, date=date)
+
+
+def try_import_wikiextractor():
+    try:
+        import WikiExtractor
+    except ImportError:
+        try:
+            download(
+                'https://raw.githubusercontent.com/attardi/wikiextractor/'
+                '16186e290d9eb0eb3a3784c6c0635a9ed7e855c3/WikiExtractor.py',
+                path=os.path.join(_CURR_DIR, 'WikiExtractor.py'),
+                sha1_hash='3c4896a837b75c476d23c037e8d6c7fdfd9a29eb')
+            import WikiExtractor
+        except:
+            raise ImportError('Cannot import WikiExtractor! You can download the "WikiExtractor.py"'
+                              ' in https://github.com/attardi/wikiextractor to {}'
+                              .format(_CURR_DIR))
+    return WikiExtractor
+
+
+class WikicorpusTextFormatting:
+    def __init__(self, wiki_path, output_filename, recursive=False):
+        self.wiki_path = wiki_path
+        self.recursive = recursive
+        self.output_filename = output_filename
+
+    # This puts one article per line
+    def merge(self):
```

Comment:
I recently just noted that this wiki cleaner does not remove some useless documents containing only html tags such as "colspan="2" style="background-color:".

Reply:
And, these tags appear in all wiki texts for all languages.

Reply:
I see. We should improve this.

Reply:
@szhengac  How about solve it in another PR?